### PR TITLE
WIP: New package: ghidra-9.0.4

### PR DESCRIPTION
**Pull request by knusbaum**
Now that openjdk11 is in, we can package this.
I'm looking for comments on this template because it's kind of messy. I haven't found a prettier way to do it.

### CODE REVIEW

File: srcpkgs/ghidra/template [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+allprojects {
+	repositories {
+		mavenCentral()
+		jcenter()
+		flatDir name:'flat', dirs:["$HOME/flatRepo"]
+	}
+}
+EOF
+}
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]: Imo this should be cleaned up after do_install.

File: srcpkgs/ghidra/template [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Template file for 'ghidra'
+pkgname=ghidra
+version=9.0.4
+revision=1
+wrksrc=ghidra
+create_wrksrc=yes
+hostmakedepends="gradle curl unzip bison flex openjdk-11"
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]:
```suggestion
hostmakedepends="gradle curl unzip bison flex openjdk11"
```

File: srcpkgs/ghidra/template [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+makedepends=""
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]:
```suggestion
```

File: srcpkgs/ghidra/template [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+depends="openjdk-11"
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]:
```suggestion
depends="openjdk11"
```

File: srcpkgs/ghidra/template [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	cd ghidra-Ghidra_9.0.4_build
+	gradle buildGhidra
+}
+
+do_install() {
+	vlicense LICENSE.txt
+
+	mkdir ghidra_dist
+	cd ghidra_dist
+	unzip "../ghidra-Ghidra_${version}_build/build/dist/ghidra*.zip"
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]:
```suggestion
	unzip -o "../ghidra-Ghidra_${version}_build/build/dist/ghidra*.zip"
```
So it doesn't fail on reruns

File: srcpkgs/ghidra/template [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+do_build() {
+	cd ghidra-Ghidra_9.0.4_build
+	gradle buildGhidra
+}
+
+do_install() {
+	vlicense LICENSE.txt
+
+	mkdir ghidra_dist
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]:
```suggestion
	mkdir -p ghidra_dist
```
So it doesn't fail on reruns

File: srcpkgs/ghidra/template [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+	cd ghidra_dist
+	unzip "../ghidra-Ghidra_${version}_build/build/dist/ghidra*.zip"
+	vmkdir usr/lib/ghidra
+	vmkdir usr/bin
+	vcopy "ghidra*/*" usr/lib/ghidra/
+	ln -s /usr/lib/ghidra/ghidraRun ${DESTDIR}/usr/bin/ghidra
```
Comment (Johnnynator) [inline srcpkgs/ghidra/template]:
```suggestion
	ln -sf /usr/lib/ghidra/ghidraRun ${DESTDIR}/usr/bin/ghidra
```
So it doesn't fail on reruns

**Comment by Johnnynator** [comment]
Add `archs="x86_64*"` gradle errors out for anything else (`/ghidra/ghidra-Ghidra_9.0.4_build/build.gradle:283`)

**Comment by abenson** [comment]
@knusbaum I am using this template for 9.1. It seems to work well for me. Doesn't need some of the patching yours did for 9.0.4. 

https://github.com/void-linux/void-packages/compare/master...abenson:np_ghidra_src

**Comment by knusbaum** [comment]
@abenson That sounds good.
I will close this since I haven't had the time to work on it that I hoped I would.

**Pull request closed without merging**